code/espurna/sensors/INA219Sensor.h

```diff
@@ -377,6 +381,15 @@ class INA219Sensor : public I2CSensor<BaseEmonSensor> {
         _dirty = true;
     }
 
+    void setShuntRes(float shunt) {
+        _shunt_Factor = shunt / 0.1;
+        _dirty = true;
+    }
+
+    void setCorrectionFactor(float corr){
+        corr_calibration = _params.calibration * corr;
+    }
```

Comment:
See datasheet, plus my excerpt from it in the comment.
$cal = \frac{0.04096}{current LSB * R shunt}$
$current LSB = \frac{maximum expected current}{2^{15}}$
$power LSB = 20 * current LSB$

Expected current is based on PGA gain, if I understood the values correctly.
Why not apply it like that so we have every value correctly scaled, keeping the structure?
(and yeah, existing lsb values are also off, looking at the table in the comment I mentioned... mA vs. A, and I think the calc is in milli scale)